modules/core/test/unit/bip32path.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @prettier
+ */
+import 'should';
+import * as bip32 from 'bip32';
+import { fromLegacyPath } from '../../src/bip32path';
+import { getSeed } from '../lib/keys';
+import { Derivable, hdPath } from '../../src/bitcoin';
+
+const utxolib = require('@bitgo/utxo-lib');
+
+describe('bip32util', function () {
+  const seed = getSeed('bip32test');
+
+  const testKeyBip32Master = bip32.fromSeed(seed);
+  const testKeyBip32Child = testKeyBip32Master.derivePath('m/0');
+
+  const testKeyHDPathMaster = utxolib.HDNode.fromSeedBuffer(seed);
+  const testKeyHDPathChild = testKeyHDPathMaster.derivePath('m/0');
+
+  /**
+   * Asserts that all inputs convert to the same reference path.
+   * Asserts that
+   *    refBIP32.derivePath(refPath) === refHDPath.derive(fromLegacyPath(p))
+   *
+   * @param inputs - array of paths that are equivalent. The first element is the reference path.
+   * @param refBIP32 - bip32 node
+   * @param refHDPath - `hdPath()` helper node
+   */
+  function runTest(inputs: string[], refBIP32: bip32.BIP32Interface, refHDPath: Derivable) {
+    if (!refHDPath) {
+      throw new Error(`invalid state`);
+    }
+
+    if (inputs.length === 0) {
+      throw new Error(`invalid argument`);
+    }
+
+    const refPath = inputs[0];
+
+    fromLegacyPath(refPath).should.eql(refPath);
```

Comment:
This comparison check is happening a couple lines below as part of the `forEach` loop I believe - I figure this line can be removed so it only happens once?

Reply:
the stack trace for this error is slightly clearer, but I should instead improve the error on the other assertion